cmake/GetGitCommitHash.cmake

```diff
@@ -1,31 +1,11 @@
 function(get_git_commit_hash)
-    get_filename_component(GIT_DESCRIBE_CMAKE_DIR ${CMAKE_CURRENT_LIST_FILE} PATH)
-    get_filename_component(GIT_ROOT ${GIT_DESCRIBE_CMAKE_DIR} PATH)
-    set(GIT_DIR "${GIT_ROOT}/.git")
+    execute_process(COMMAND git rev-parse --git-dir OUTPUT_VARIABLE GIT_DIR OUTPUT_STRIP_TRAILING_WHITESPACE)
+    execute_process(COMMAND git rev-parse HEAD OUTPUT_VARIABLE HEAD_REF OUTPUT_STRIP_TRAILING_WHITESPACE)
 
     # Add a CMake configure dependency to the currently checked out revision.
     set(GIT_DEPENDS ${GIT_DIR}/HEAD)
-    file(READ ${GIT_DIR}/HEAD HEAD_REF)
-    if(HEAD_REF MATCHES "ref: (.+)\n")
-        set(HEAD_REF ${CMAKE_MATCH_1})
-        if(EXISTS "${GIT_DIR}/${HEAD_REF}")
-            list(APPEND GIT_DEPENDS ${GIT_DIR}/${HEAD_REF})
-            file(READ ${GIT_DIR}/${HEAD_REF} HEAD_REF)
-        elseif(EXISTS "${GIT_DIR}/packed-refs")
-            list(APPEND GIT_DEPENDS ${GIT_DIR}/packed-refs)
-            file(READ "${GIT_DIR}/packed-refs" PACKED_REFS)
-            if(${PACKED_REFS} MATCHES "([0-9a-z]*) ${HEAD_REF}")
-                set(HEAD_REF ${CMAKE_MATCH_1})
-            else()
-                set(HEAD_REF "")
-            endif()
-        else()
-            set(HEAD_REF "")
-        endif()
-    endif()
     set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${GIT_DEPENDS})
 
-    string(STRIP ${HEAD_REF} HEAD_REF)
```

Comment:
this is redundant since `execute_process`'s `OUTPUT_STRIP_TRAILING_WHITESPACE` option takes care of trailing whitespace and the output of  `git rev-parse` doesn't containg leading whitespace. But: if `git rev-parse` fails, `${HEAD_REF}` is undefined and cmake gives a confusing error message regarding the number of arguments give to `strip`